utt/wallet-cli/src/fileBasedUserStorage.cpp

```diff
@@ -0,0 +1,167 @@
+// UTT Client API
+//
+// Copyright (c) 2020-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#if __has_include(<filesystem>)
+#include <filesystem>
+namespace fs = std::filesystem;
+#elif __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+#else
+#error "Missing filesystem support"
+#endif
+#include <utt/Serialization.h>
+#include <fstream>
+#include <sstream>
+#include <xutils/Log.h>
+#include "fileBasedUserStorage.hpp"
+#include "serialization.hpp"
+
+namespace utt::client {
+std::string bytesToHex(const std::string& bytes) {
+  std::stringstream s;
+  for (size_t i = 0; i < bytes.size(); i++) {
+    // Convert from signed char to std::uint8_t and then to an unsigned non-char type so that it prints as an integer.
+    const auto u = static_cast<std::uint8_t>(bytes[i]);
+    s << std::hex << std::setw(2) << std::setfill('0') << static_cast<std::uint16_t>(u);
+  }
+  return s.str();
+}
+
+std::string bytesToHex(const std::vector<uint8_t>& bytes) {
+  std::stringstream s;
+  for (size_t i = 0; i < bytes.size(); i++) {
+    s << std::hex << std::setw(2) << std::setfill('0') << static_cast<std::uint16_t>(bytes[i]);
+  }
+  return s.str();
+}
+
+std::vector<uint8_t> hexStringToBytes(const std::string& hex) {
+  std::vector<uint8_t> bytes;
+
+  for (unsigned int i = 0; i < hex.length(); i += 2) {
+    std::string byteString = hex.substr(i, 2);
+    uint8_t byte = (uint8_t)strtol(byteString.c_str(), NULL, 16);
+    bytes.push_back(byte);
+  }
+
+  return bytes;
+}
+
+FileBasedUserStorage::FileBasedUserStorage(const std::string& path)
+    : state_path_{path + "/.state.json"}, pending_path_{path + "/.pending.json"}, lock_path_{path + "/.LOCK"} {
+  fs::create_directories(path);
+  if (fs::exists(lock_path_)) {
+    // If we have a lock file, then we have a pending file that we need to write to the actual storage.
+    if (!fs::exists(pending_path_)) throw std::runtime_error("storage is corrupted");
+    fs::copy(pending_path_, state_path_, fs::copy_options::overwrite_existing);
+    fs::remove(lock_path_);
+    fs::remove(pending_path_);
+  }
+  if (fs::exists(state_path_)) {
+    std::ifstream f(state_path_);
+    state_->state_ = json::parse(f);
+  }
+}
+
+void FileBasedUserStorage::startTransaction() {
+  if (!state_->state_.contains("initialized")) state_->state_["initialized"] = true;
```

Comment:
what is the meaning of ```startTransaction``` ? it just set the state to initialized ?

Reply:
Shouldn't this be clear from the method's documentation in the header file?